### PR TITLE
Improve error message on partial failure

### PR DESCRIPTION
**Pull request by mandolaerik**

### CODE REVIEW

**Comment by syssimics** [comment]
DML [#9818414](https://teamcity01-ir.devtools.intel.com/viewLog.html?buildId=9818414): fail

**Comment by syssimics** [comment]
Verification [#9818413](https://teamcity01-ir.devtools.intel.com/viewLog.html?buildId=9818413): fail

File: py/port_dml.py [hunk 1]
```diff
@@ -1050,19 +1050,23 @@ def main(argv):
...
 
     PWUNUSED.report(src, args.dest)
 
     with (open(args.dest, 'w', newline='') if args.dest else sys.stdout) as f:
         src.commit(f)
-    if fail:
+    if errors:
+        total = sum(map(len, transformations.values()))
+        sys.stderr.write(f'''\
+*** Failed to apply {errors} out of {total} porting tags; partial result saved to {args.dest}. Consider applying the failed tags manually.
```
Comment (svenwest) [inline py/port_dml.py:1068]: Great, how would one apply these tags manually?

Comment (mandolaerik) [reply py/port_dml.py:1068]: After injecting a bug in the porting script, the failure is presented as follows:
```
Unexpected error on this porting tag:
/disk1/ecarsten/6/src/tools/dmlc/test/1.2/misc/porting.dml:35:23: porting PHARD_RESET_VALUE: []
/disk1/ecarsten/6/logs/test/t126_dmlc/scratch/porting/tags:52: found here
Traceback (most recent call last):
  File "/disk1/ecarsten/6/linux64/bin/dml/python/port_dml.py", line 1048, in main
    t.apply(src)
  File "/disk1/ecarsten/6/linux64/bin/dml/python/port_dml.py", line 334, in apply
    assert act_before == exp_before, act_before
AssertionError: hard_reset_value
```
Looking at porting.dml:35, I see:
```
            parameter hard_reset_value = 3;
```
Looking at the same place in the generated file (now line 37), I see:
```
            param hard_reset_value = 3;
```
Now, if I search the manual for PHARD_RESET_VALUE, I find:

> The `hard_reset_value` parameter is no longer
> recognized. The parameter is automatically renamed to
> `init_val`, which has roughly the same effect.

so I can change line 37 to:
```
            param init_val = 3;
```

Comment (svenwest) [reply py/port_dml.py:1068]: Ok, I was just wondering how the wording `applying the failed tags` come into play here, can I correlate the errors to the tags file and see what needs to be ported?

Comment (mandolaerik) [reply py/port_dml.py:1068]: yes, please re-read my comment. If the final message says there are N errors, then it is preceded by N tracebacks on the mentioned form, the location in the tag file is mentioned, and that line is also displayed (`tags:52` reads `/disk1/ecarsten/6/src/tools/dmlc/test/1.2/misc/porting.dml:35:23: porting PHARD_RESET_VALUE: []`)

Comment (svenwest) [reply py/port_dml.py:1068]: Ah, got it. Sorry for the noise.

**Review by svenwest — APPROVED** [review]

**Pull request merged**